cladecombiner/ordered_taxon.py

```diff
@@ -0,0 +1,42 @@
+from functools import total_ordering
+from typing import Any
+
+from .taxon import Taxon
+from .taxonomy_scheme import TreelikeTaxonomyScheme
+
+
+@total_ordering
+class OrderedTaxon(Taxon):
+    """
+    A Taxon equipped with a TreelikeTaxonomyScheme which can be sorted.
+
+    For the purposes of comparison, "A < B" means "A is contained within B (but is not B)."
```

Comment:
So nit-picky, but I would think it would be easier to have `A in B`

Then you would need to sort with `sorted(my_taxa, key=lambda x, y: x in y)` or something, which maybe is annoying, but which I find much clearer

Reply:
I'm not opposed, "in" is the sensible terminology here.

Still new to all of python's "dunder" methods, The Internet tells me that to do this I would need to define `OrderedTaxon.__contains__()`?

Reply:
I have removed this class in favor of a new `taxon_utils.sort_taxa()` function which uses `sorted` in #13